### PR TITLE
Reader sidebar: switch from lib/user-settings to Redux

### PR DESCRIPTION
**Pull request by bluefuton**
The Reader sidebar renders a promo for the WordPress desktop app, and currently uses `lib/user-settings` to find the user's locale and check whether they're already a desktop app user.

This PR switches to use Redux and also refactors the promo into to its own component.

### To test

Using an account in the `en` locale, check that the desktop promo appears in your sidebar.

screen shot 2018-07-20 at 16 20 43

### CODE REVIEW

File: client/reader/sidebar/promo.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	const {
+		isDesktopPromoDisabled = store.get( 'desktop_promo_disabled' ),
+		isViewportMobile = isMobile(),
+		isUserLocaleEnglish = 'en' === options.currentUserLocale,
+		isDesktopPromoConfiguredToRun = config.isEnabled( 'desktop-promo' ),
+		isUserDesktopAppUser = haveUserSettingsLoaded || options.isDesktopAppUser,
+		isUserOnChromeOs = /\bCrOS\b/.test( navigator.userAgent ),
+	} = options;
+
+	return every( [
```
Comment (blowery) [inline client/reader/sidebar/promo.jsx]: why are you using `every` here instead of an `&&` chain?

**Comment by blowery** [comment]
Woo hoo!

**Comment by blowery** [comment]
Is the "desktop" usage here just a misnomer? This appears to be a promo for the native apps, not the desktop app.

**Comment by blowery** [comment]
> Is the "desktop" usage here just a misnomer? This appears to be a promo for the native apps, not the desktop app.

Ah no, I see a desktop nudge. My mistake.

**Review by blowery — APPROVED** [review]
LGTM. Feel free to ignore my nit on `every` usage.

File: client/reader/sidebar/promo.jsx (hunk 1 above)
Comment (bluefuton) [reply client/reader/sidebar/promo.jsx]: We'll have to ask the original author :) I just ported across from `sidebar/index.jsx`.

Worth cleaning up though - I'll fix now.

**Pull request merged**